src/Castle.Windsor/Core/ComponentModel.cs

```diff
@@ -331,6 +332,12 @@ public IEnumerable<Type> Services
 		{
 			get { return services; }
 		}
+		
+		[DebuggerDisplay("Count = {services.Count}")]
+		public HashSet<Type> FastServices
+		{
+			get { return servicesFast; }
+		}
```

Comment:
This is on the public interface, I don't think that is right. What sort of performance improvement are you seeing?

Could ComponentModel just have a ContainsService (or similar name) method so the internals aren't exposed?

Reply:
I agree that this should not be part of the public api. I just removed it from there. I just internalized the collection for now. Exposing the collection itself internally seems necessary. For that look into [ComponentServicesUtil](https://github.com/sqeezy/Windsor/blob/faster_type_lookup/src/Castle.Windsor/Core/Internal/ComponentServicesUtil.cs#L25).

The root of my problems is in [AbstractHandler](https://github.com/sqeezy/Windsor/blob/faster_type_lookup/src/Castle.Windsor/MicroKernel/Handlers/AbstractHandler.cs#L186). When registering a lot of components the main effort of our application build goes into the Contains implementation of List<T>, which does not scale well. That effort is reduced to almost none with a parallel HashSet<T> in use.